common/src/main/java/com/microsoft/identity/common/internal/commands/RefreshOnCommand.java

```diff
@@ -0,0 +1,72 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.commands;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.exception.ServiceException;
+import com.microsoft.identity.common.internal.commands.parameters.CommandParameters;
+import com.microsoft.identity.common.internal.commands.parameters.SilentTokenCommandParameters;
+import com.microsoft.identity.common.internal.controllers.BaseController;
+import com.microsoft.identity.common.internal.result.EmptyResult;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.io.IOException;
+
+public class RefreshOnCommand extends BaseCommand<EmptyResult>{
+
+    private static final String TAG = SilentTokenCommand.class.getSimpleName();
+
+    public RefreshOnCommand(@NonNull CommandParameters parameters, @NonNull BaseController controller, @NonNull String publicApiId) {
+        super(parameters, controller, new EmptyCallback(), publicApiId);
+    }
+
+    @Override
+    public EmptyResult execute() throws IOException, ClientException, ServiceException {
+        final String methodName = ":execute";
+
+        final BaseController controller = getDefaultController();
+        Logger.verbose(
+                TAG + methodName,
+                "Executing with controller: "
+                        + controller.getClass().getSimpleName()
+        );
+        SilentTokenCommandParameters commandParameters = (SilentTokenCommandParameters) getParameters();
+        controller.renewAccessToken(commandParameters);
+
+        return new EmptyResult();
```

Comment:
Ideally we would just do Void

Reply:
It's because execute() has a return type, which means if we override it we have to provide an object. This is better than returning null, but in order to return void we'd have to update either BaseCommand or not extend it.

Reply:
`Void` is different than `void` - with that said, I'm actually thinking we should make this method return an actual result..I think the command should return the result of its operation...it is just that we are going to ignore that in the Disptacher for now.

Reply:
But we are supposed to forget this result though, right? :)